### DISCUSSION TITLE
How does Stockfish determine how long to think on moves?

### DISCUSSION THREAD

**Discussion by Clausable** [official-stockfish/Stockfish#4292] [Q&A] [answered]
I'm just running a chess engine tournament on my computer just for fun, and I was wondering **how does Stockfish determine how long to think on moves?** I noticed that sometimes Stockfish thinks like 5 seconds on the first move during 1+0 time control, and then starts spending less than a second on the next several moves. Why and how does Stockfish do this?

**Comment by Sopel97** [comment]
It's complicated

https://github.com/official-stockfish/Stockfish/blob/c2d507005c51145211a7963af7c41026bd759d08/src/timeman.cpp#L37-L103
https://github.com/official-stockfish/Stockfish/blob/c2d507005c51145211a7963af7c41026bd759d08/src/search.cpp#L458-L494

and a few more places in search

**Comment by vondele** [comment, marked as answer]
A relatively simple scheme establishes the base time (more or less a fraction of the available time), and all kind of factors such as bestmove stability, complexity of the position, stability of the evaluation, etc. influence how much this base time is adjusted to the needs of the position.
